datasync/fetcher/fetcher.go

```diff
@@ -340,6 +341,9 @@ func (f *Fetcher) loop() {
 				if f.queueChangeHook != nil {
 					f.queueChangeHook(hash, true)
 				}
+				insertFailed = true
+			}
+			if insertFailed {
```

Comment:
how about this?
```
L:
for hash, announce := range f.fetching {
    ...
    select {
        ...
        break L
    }
```

Reply:
That would be better! I will update the change.